string key
string value
string min
string max